nvpair/src/lib.rs

```diff
@@ -372,6 +372,49 @@ impl NvEncode for str {
     }
 }
 
+impl NvEncode for &[u64] {
```

Comment:
We already have `impl NvEncode for [u64]`.  Do we need both?  I'm not totally clear on when we need it for `&[u64]` vs `[u64]` - in insert_into(), I think the type of `self` is `&&[u64]`.  Same applies to the routines below, e.g. `&[&str]` vs `[&str]`.